dask_planner/src/expression.rs

```diff
@@ -146,13 +179,30 @@ impl PyExpr {
                         Ok(fq_name) => {
                             let mut idx: usize = 0;
                             for schema in input_plans[0].all_schemas() {
-                                match schema.index_of_column(&Column::from_qualified_name(&fq_name))
-                                {
-                                    Ok(e) => {
-                                        idx = e;
-                                        break;
+                                let (schema_name_opt, column_name) =
+                                    self._schema_column_from_fqn(&self.expr, &fq_name);
+                                match schema_name_opt {
+                                    Some(e) => {
+                                        match schema.index_of_column_by_name(
+                                            Some(e.as_str()),
+                                            column_name.as_str(),
+                                        ) {
+                                            Ok(e) => {
+                                                idx = e;
+                                                break;
+                                            }
```

Comment:
We can simplify the flow and just return here:

```suggestion
                                            Ok(idx) => return Ok(idx),
```

Reply:
Resolved in https://github.com/dask-contrib/dask-sql/pull/732